fabric-provider-api-v1/src/main/java/net/fabricmc/fabric/api/provider/v1/ApiProviderAccess.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.provider.v1;
+
+import net.fabricmc.fabric.impl.provider.ApiProviderAccessRegistry;
+
+/**
+ * Describes and provides access to component instances that may be retrieved
+ * for blocks, items or entities.
```

Comment:
Maybe we should clarify that there are default access providers for blocks items and entities. This current sounds like only blocks, block entities, items and entities are supported.

For example, Waterpicker has pressed interest in possibly having a `Rift` from dimensional doors be a provider of api access. Hence why I want to have the javadoc here to not explicitly undersell the fact the api may be used for more than just built-in vanilla types.

Reply:
agree